spec/unit/lib/temporal/client/retryer.rb

```diff
@@ -0,0 +1,45 @@
+require 'temporal/client/retryer'
+require 'temporal/metadata'
+require 'time'
+
+describe Temporal::Client::Retryer do
+  it 'backs off and stops retrying eventually' do
+    timestamps = []
+    max_wait = 5.0
+    expect do
+      result = described_class.retry_for(max_wait, retry_message: "Still trying", metadata_hash: {}) do
+        timestamps << Time.now.to_f
+        raise 'try again'
+      end
+    end.to raise_error(StandardError)
+    expect(timestamps.last - timestamps.first).to be <= max_wait
+
+    # Test backoff
+    initial_interval = 0.2
+    backoff = 1.2
```

Comment:
Should these constants be read from the main code instead of redefined?

Reply:
@jeffschoner-stripe I tend to _not_ do that since the point is to provide a check on the production code.  People should have to look at the test and see if they intend the change.